akka-http-tests/src/test/java/akka/http/javadsl/server/MinimalHttpApp.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.javadsl.server;
+
+import akka.actor.ActorSystem;
+import scala.runtime.BoxedUnit;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CompletableFuture;
+
+public class MinimalHttpApp extends HttpApp {
+
+  CompletableFuture<BoxedUnit> shutdownTrigger = new CompletableFuture<>();
```

Comment:
show in the app that you trigger this somehow, add a route that does `/shutdown` and triggers this.
Otherwise some people will certainly be confused ;)